native/spark-expr/src/conversion_funcs/cast.rs

```diff
@@ -866,6 +867,40 @@ pub fn spark_cast(
     }
 }
 
+// copied from datafusion common scalar/mod.rs
+fn dict_from_values<K: ArrowDictionaryKeyType>(
```

Comment:
should we make it `pub` in DataFusion and reuse ?

Reply:
we can also play with `cast` kernel, looks like such cast might be supported

from `can_cast_types`
```
        (_, Dictionary(_, value_type)) => can_cast_types(from_type, value_type),
```

Reply:
That's a good suggestion. Let me try that.

Reply:
I looked into reusing the cast kernel. Looks like existing code already would have fallen thru to the catch all branch which calls the cast kernel. Even though `can_cast_types` appears to allow this, the cast actually fails with 
```
Caused by: org.apache.comet.CometNativeException: InternalError: Native cast invoked for unsupported cast from Utf8 to Dictionary(Int32, Utf8).
```

The reason for the cast (which is implicit and does not show up in the plan) is apparently because of a case statement like 
```
CASE WHEN (_condition_) THEN _dict_encoded_column_ ELSE '' END
```
In such a case, we are implicitly trying to cast the array containing the string value to a dict encoded array.

I'll open a follow up PR to make `dict_from_values` publich in DF and remove the duplication

Reply:
@mbutrovich @comphead can we go ahead with this PR? I've verified that using the cast kernel directly does not work for this case.